tests/wallet/nft_wallet/test_nft_offers.py

```diff
@@ -149,12 +159,28 @@ async def test_nft_offer_with_fee(
     peer = wallet_node_1.get_full_node_peer()
     assert peer is not None
     trade_take, tx_records = await trade_manager_taker.respond_to_offer(
-        old_maker_offer if forwards_compat else Offer.from_bytes(trade_make.offer), peer, fee=taker_fee
+        old_maker_offer if forwards_compat else Offer.from_bytes(trade_make.offer),
+        peer,
+        fee=taker_fee,
+        reuse_puzhash=reuse_puzhash and not forwards_compat,
     )
 
     assert trade_take is not None
     assert tx_records is not None
-
+    if reuse_puzhash and not forwards_compat:
```

Comment:
you should also add an `else` condition here that checks derivation records have changed in case `reuse_puzhash=False`